appcues/src/main/java/com/appcues/ui/ExperienceRenderer.kt

```diff
@@ -1,50 +1,42 @@
 package com.appcues.ui
 
 import com.appcues.AppcuesConfig
-import com.appcues.AppcuesCoroutineScope
 import com.appcues.SessionMonitor
-import com.appcues.analytics.AnalyticsEvent
-import com.appcues.analytics.AnalyticsTracker
+import com.appcues.action.ExperienceAction
+import com.appcues.analytics.ExperienceLifecycleEvent.StepInteraction.InteractionType
 import com.appcues.analytics.ExperienceLifecycleTracker
 import com.appcues.data.AppcuesRepository
 import com.appcues.data.model.Experience
 import com.appcues.data.model.ExperiencePriority.NORMAL
 import com.appcues.data.model.ExperienceTrigger
-import com.appcues.data.model.Experiment
+import com.appcues.data.model.RenderContext
 import com.appcues.statemachine.Action.EndExperience
 import com.appcues.statemachine.Action.StartExperience
 import com.appcues.statemachine.Action.StartStep
 import com.appcues.statemachine.Error
+import com.appcues.statemachine.Error.NoActiveStateMachine
 import com.appcues.statemachine.State
 import com.appcues.statemachine.State.Idling
 import com.appcues.statemachine.StateMachine
 import com.appcues.statemachine.StateMachine.StateMachineListener
+import com.appcues.statemachine.StateMachineFactory
 import com.appcues.statemachine.StepReference
 import com.appcues.util.ResultOf
 import com.appcues.util.ResultOf.Failure
 import com.appcues.util.ResultOf.Success
-import com.appcues.util.appcuesFormatted
 import kotlinx.coroutines.flow.MutableSharedFlow
 import kotlinx.coroutines.flow.SharedFlow
-import kotlinx.coroutines.launch
-import org.koin.core.component.KoinScopeComponent
-import org.koin.core.component.inject
-import org.koin.core.scope.Scope
+import kotlin.collections.set
 
 internal class ExperienceRenderer(
-    override val scope: Scope,
-    private val appcuesCoroutineScope: AppcuesCoroutineScope,
+    private val config: AppcuesConfig,
     private val experienceLifecycleTracker: ExperienceLifecycleTracker,
-) : KoinScopeComponent, StateMachineListener {
+    private val repository: AppcuesRepository,
+    private val sessionMonitor: SessionMonitor,
+    private val stateMachineFactory: StateMachineFactory,
+) : StateMachineListener {
 
-    private val repository by inject<AppcuesRepository>()
-    private val stateMachine by inject<StateMachine>()
-    private val sessionMonitor by inject<SessionMonitor>()
-    private val config by inject<AppcuesConfig>()
-
-    // lazy prop inject here to avoid circular dependency with AnalyticsTracker
-    // AnalyticsTracker > AnalyticsQueueProcessor > ExperienceRenderer(this) > AnalyticsTracker
-    private val analyticsTracker by inject<AnalyticsTracker>()
+    private val stateMachines: HashMap<RenderContext, StateMachine> = hashMapOf()
```

Comment:
will need something like this. Im not yet following how we'll track what `AppcuesFrame` (view) owns each state machine for the embed views, and do so in a way that allows for memory to be managed when those views go out of scope. See the iOS side where there is the one `stateMachine` in the `ExperienceRenderer`, then the `StateMachineDirectory` that manages this mapping and lookup with weak references to the `StateMachineOwning` (views)